shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.mm

```diff
@@ -491,3 +447,271 @@ - (void)insertViewToPoolIfNeeded:(FlutterClippingMaskView*)maskView {
 }
 
 @end
+
+@implementation UIView (FirstResponder)
+- (BOOL)flt_hasFirstResponderInViewHierarchySubtree {
+  if (self.isFirstResponder) {
+    return YES;
+  }
+  for (UIView* subview in self.subviews) {
+    if (subview.flt_hasFirstResponderInViewHierarchySubtree) {
+      return YES;
+    }
+  }
+  return NO;
+}
+@end
+
+// This recognizers delays touch events from being dispatched to the responder chain until it failed
+// recognizing a gesture.
+//
+// We only fail this recognizer when asked to do so by the Flutter framework (which does so by
+// invoking an acceptGesture method on the platform_views channel). And this is how we allow the
+// Flutter framework to delay or prevent the embedded view from getting a touch sequence.
+@interface DelayingGestureRecognizer : UIGestureRecognizer <UIGestureRecognizerDelegate>
```

Comment:
IMO this should be "namespaced". This looks like a common class name and we'll end up with collisions in some cases.

Reply:
Done